util/types/convert.go

```diff
@@ -171,9 +171,11 @@ func StrToDateTime(str string, fsp int) (Time, error) {
 	return ParseTime(str, mysql.TypeDatetime, fsp)
 }
 
-// StrToDuration converts str to Duration.
+// StrToDuration converts str to Duration. It returns Duration in normal case,
+// and returns Time when str is in datetime format.
+// when isDuration is true, the d is returned, when it is false, the t is returned.
 // See https://dev.mysql.com/doc/refman/5.5/en/date-and-time-literals.html.
-func StrToDuration(sc *variable.StatementContext, str string, fsp int) (t Time, err error) {
+func StrToDuration(sc *variable.StatementContext, str string, fsp int) (d Duration, t Time, isDuration bool, err error) {
```

Comment:
Str2TimeOrDuration better?

Reply:
MySQL use strToTime, so I think no need to change it...the better way is to eliminate the duration type totally.